### PR TITLE
feat: Add AI insight component scraper

### PR DESCRIPTION
**Pull request by zreigz**

### CODE REVIEW

File: pkg/scraper/component_scraper.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	}
+}
+
+func setUnhealthyComponents(ctx context.Context, k8sClient ctrclient.Client, gvk schema.GroupVersionKind, opts ...ctrclient.ListOption) error {
+	// Create an unstructured list with the desired GVK
+	list := &unstructured.UnstructuredList{}
+	list.SetGroupVersionKind(gvk)
+
+	// List resources
+	if err := k8sClient.List(ctx, list, opts...); err != nil {
```
Comment (michaeljguarino) [inline pkg/scraper/component_scraper.go:72]: we should definitely paginate the list here if we can (don't think it should take much of a refactor).  Mostly to reduce a lot of pressure on the k8s apiserver

File: pkg/ping/build.go [hunk 2]
```diff
@@ -3,16 +3,42 @@ package ping
...
 	console "github.com/pluralsh/console/go/client"
 	"github.com/samber/lo"
 	"k8s.io/apimachinery/pkg/version"
 )
 
 func pingAttributes(info *version.Info, pods []string, minKubeletVersion *string) console.ClusterPing {
 	vs := strings.Split(info.GitVersion, "-")
+	var insightComponents []*console.ClusterInsightComponentAttributes
+	if !scraper.AiInsightComponents.IsEmpty() {
+		insightComponents = make([]*console.ClusterInsightComponentAttributes, 0)
```
Comment (michaeljguarino) [inline pkg/ping/build.go:17]: minor things, if we send an empty list, it'll actually wipe the components on the server, so if empty just don't add to the cluster ping request at all (I'm hoping it has omitempty on it).  Also i think the full insight list should be fresh or not fresh, not individual ones, that way you can control easily whether to add the whole thing to the ping request.

File: pkg/scraper/component_scraper.go (hunk 1 above)
Comment (zreigz) [reply pkg/scraper/component_scraper.go]: done, I have used our `algorithms.NewPager`for it

**Comment by zreigz** [comment]
@michaeljguarino PTAL

**Review by michaeljguarino — APPROVED** [review]
overall lgtm, one thing, I added a ticket, we should add node to this list (might need a direct health implementation), but it would be super useful to get health insights for those objects as well to get a heads up on node health issues.

**Pull request merged**